tests/php/Shortcodes/ImageShortcodeProviderTest.php

```diff
@@ -149,4 +150,37 @@ public function testMissingImageDoesNotCache()
         // assert that cache is still empty after parsing shortcode
         $this->assertNull($cache->get($cacheKey));
     }
+
+    public function testLazyLoading()
+    {
+        $parser = new ShortcodeParser();
+        $parser->register('image', [ImageShortcodeProvider::class, 'handle_shortcode']);
+
+        $id = $this->objFromFixture(Image::class, 'imageWithTitle')->ID;
+
+        // regular shortcode
+        $shortcode = '[image id="' . $id . '" width="300" height="200"]';
+        $this->assertContains('loading="lazy"', $parser->parse($shortcode));
+
+        // missing width
+        $shortcode = '[image id="' . $id . '" height="200"]';
+        $this->assertNotContains('loading="lazy"', $parser->parse($shortcode));
+
+        // missing height
+        $shortcode = '[image id="' . $id . '" width="300"]';
+        $this->assertNotContains('loading="lazy"', $parser->parse($shortcode));
+
+        // loading="eager"
+        $shortcode = '[image id="' . $id . '" width="300" height="200" loading="eager"]';
```

Comment:
Maybe add a case where some non-sense gets put in the loading attribute

Reply:
I have added nonsense